core/lib/types/src/storage_writes_deduplicator.rs

```diff
@@ -101,10 +105,16 @@ impl StorageWritesDeduplicator {
             };
 
             let is_write_initial = log.log_type == StorageLogQueryType::InitialWrite;
-            let field_to_change = if is_write_initial {
-                &mut self.metrics.initial_storage_writes
+            let (field_to_change, total_size) = if is_write_initial {
+                (
+                    &mut self.metrics.initial_storage_writes,
+                    &mut self.metrics.total_writes_size,
```

Comment:
In both `if` and `else` clauses you return the same `&mut self.metrics.total_writes_size`, maybe we could do it without it and simplify the code